core/src/main/scala/org/bitcoins/core/protocol/script/ScriptSignature.scala

```diff
@@ -248,16 +248,32 @@ object P2SHScriptSignature extends ScriptFactory[P2SHScriptSignature] {
   /** Detects if the given script token is a redeem script */
   def isRedeemScript(token: ScriptToken): Boolean = {
     val redeemScript: ScriptPubKey = parseRedeemScript(token)
-    redeemScript match {
-      case _: P2PKHScriptPubKey | _: MultiSignatureScriptPubKey |
-          _: P2SHScriptPubKey | _: P2PKScriptPubKey |
-          _: P2PKWithTimeoutScriptPubKey | _: ConditionalScriptPubKey |
-          _: CLTVScriptPubKey | _: CSVScriptPubKey | _: WitnessScriptPubKeyV0 |
-          _: UnassignedWitnessScriptPubKey =>
-        true
-      case _: NonStandardScriptPubKey | _: WitnessCommitment => false
-      case EmptyScriptPubKey                                 => false
+
+    def isStandardNonP2SH(
+        spk: ScriptPubKey,
+        isRecursiveCall: Boolean): Boolean = {
+      spk match {
+        case _: P2PKHScriptPubKey | _: MultiSignatureScriptPubKey |
+            _: P2PKScriptPubKey | _: P2PKWithTimeoutScriptPubKey |
+            _: WitnessScriptPubKeyV0 | _: UnassignedWitnessScriptPubKey =>
+          true
+        case EmptyScriptPubKey => isRecursiveCall // Fine if nested
+        case conditional: ConditionalScriptPubKey =>
+          isStandardNonP2SH(conditional.firstSPK,
+                            isRecursiveCall = true) && isStandardNonP2SH(
+            conditional.secondSPK,
+            isRecursiveCall = true)
+        case locktime: LockTimeScriptPubKey =>
+          Try(locktime.locktime).isSuccess &&
+            isStandardNonP2SH(locktime.nestedScriptPubKey,
+                              isRecursiveCall = true)
+        case _: NonStandardScriptPubKey | _: WitnessCommitment |
+            _: P2SHScriptPubKey =>
```

Comment:
`P2SHScriptPubKey` was valid before, is this correct?

Reply:
Yes, `P2SH(P2SH(_))` is not legal in bitcoin and the only reason we didn't have a failure is because our generators don't create such a thing